movingpandas/trajectory_generalizer.py

```diff
@@ -200,3 +201,50 @@ def _generalize_traj(self, traj, tolerance):
         new_df = traj.df.iloc[keep_rows]
         new_traj = Trajectory(new_df, traj.id)
         return new_traj
+
+
+class TopDownTimeRatioGeneralizer(TrajectoryGeneralizer):
+    """
+    Generalizes using Top-Down Time Ratio algorithm proposed by Meratnia and De By (DOI: 10.1007/978-3-540-24741-8_44).
+
+    tolerance : float
+        Distance tolerance in Kilometers
+
+    Examples
+    --------
+
+    >>> mpd.TopDownTimeRatioGeneralizer(traj).generalize(tolerance=1.0)
+    """
+
+    def _generalize_traj(self, traj, tolerance):
+        # traj.reset_index(drop=True, inplace=True)
+        df = traj.df.copy()
+        # geom_col = traj.get_geom_column_name()
+
+        return Trajectory(self.td_tr(df, tolerance), traj.id)
+        
+
+    def td_tr(self, df, tolerance):
+        
+        if len(df)<=2:
+            return df
+        else:
+            de = (df.index.max().to_pydatetime() - df.index.min().to_pydatetime()).seconds
+            
+            dlat = df.geometry.iloc[-1].x - df.geometry.iloc[0].x
+            dlon = df.geometry.iloc[-1].y - df.geometry.iloc[0].y
+
+            # distances for each point and the calulated one based on it and start
+            dists = df.apply(lambda rec: self._dist_from_calced(rec, df.index.min().to_pydatetime(), df.geometry.iloc[0], de, dlat, dlon), axis=1) 
+            # print(dists)
+            # print(pd.Series(dists).idxmax())
+
+            if dists.max()>tolerance:
+                return pd.concat([self.td_tr(df.iloc[:df.index.get_loc(dists.idxmax())], tolerance), self.td_tr(df.iloc[df.index.get_loc(dists.idxmax()):], tolerance)])
+            else:
+                return df.iloc[[0,-1]]
+    
+    def _dist_from_calced(self, rec, start_t, start_geom, de, dlat, dlon):
```

Comment:
It seems like this function assumes that all trajectories are in lat/lon.

Reply:
Its the naming scheme. Will replace